lxc/remote.go

```diff
@@ -55,6 +56,35 @@ func (c *remoteCmd) flags() {
 	gnuflag.BoolVar(&c.public, "public", false, i18n.G("Public image server"))
 }
 
+func getRemoteCertificate(address string) (*x509.Certificate, error) {
+	// Setup a permissive TLS config
+	tlsConfig, err := shared.GetTLSConfig("", "", nil)
+	if err != nil {
+		return nil, err
+	}
+
+	tlsConfig.InsecureSkipVerify = true
```

Comment:
The function name is "getRemoteCertificate" and it does exactly what it says, it gets you the remote certificate.

You can't fetch the remote certificate without establishing a tls connection and since you obviously don't have it in your trust store, you have to bypass all the checks.

That function is only used in one spot in the remote command when the server certificate isn't trusted by the system CA, so we can download it and present it to the user for approval, once approved, it's stored as the certificate for that remote.